app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def new
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
```

Comment:
Because we are storing the session in the DB via the [activerecord-session_store gem](https://github.com/rails/activerecord-session_store) (I have to follow up with Amos and Justin to remind me why we are doing that), this means that when I sign out (even if I never confirmed the TOTP), this `otp_secret_key` will persist in the session, and the next user who signs in on my computer will have the exact same codes as me.

To fix this, we need to make sure to delete this session variable once it has been confirmed and [after the user signs out](https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/saml_idp_logout_concern.rb#L137-L141).

Please add a test for this.